router/src/lib.rs

```diff
@@ -849,8 +849,8 @@ pub(crate) struct ChatRequest {
 
     /// A specific tool to use. If not provided, the model will default to use any of the tools provided in the tools parameter.
     #[serde(default)]
-    #[schema(nullable = true, example = "null")]
-    pub tool_choice: ToolChoice,
+    #[schema(nullable = true, default = "null", example = "null")]
+    pub tool_choice: Option<ChatCompletionToolChoiceOption>,
```

Comment:
Why the change to `Option` you have the `NoTool` choice, there shouldn't be a need to `Option` wrapping is there ?

Reply:
`Option` is needed so the user can send a http request without a `tool_choice` param included. If `tool_choice` is `None` (user did not include a tool_choice in the request), `ChatCompletionToolChoiceOption` defaults to `NoTool`

Reply:
Doesn't a default take care of that ? And shouldn't the default be "auto"?